.travis.yml

```diff
@@ -1,16 +1,22 @@
 sudo: required
 dist: xenial
-language: python
+language: generic
+
 matrix:
     include:
-        - python: 3.7
-        - python: 3.6
-        - python: 3.5
-        - python: 2.7
```

Comment:
Thanks for the PR! Regarding Python 2.7, any particular reason why it was removed like incompatibility with the PR? Personally, I removed Python 2.7 support from all my main projects, but for this one, I think it's quite useful. I.e., I use watermark for teaching (have it on top of HW code that students submit), and I thought having Python 2.7 support would make it useful to catch the use of Python 2.7 (to explain certain bugs etc.)

Reply:
This is just about new features. You will still be able to install existing versions on Python 2.7: that's the point of the `python_requires` property.

But NEP 29 is adopted by all major Scientific Python projects. There's also https://python3statement.org/. So if you want to use the latest Scikit-learn or Numpy or Pandas or IPython, then you will need Python 3.7+ anyway, so there's no point having the maintenance burden of Python 2.7 for future releases.

Reply:
> But NEP 29 is adopted by all major Scientific Python projects. There's also https://python3statement.org/. So if you want to use the latest Scikit-learn or Numpy or Pandas or IPython, then you will need Python 3.7+ anyway, so there's no point having the maintenance burden of Python 2.7 for future releases.

Totally agree and I did that for all my scientific computing related projects. For watermark though, I do think one of the usecases is to diagnose/see what code someone is using or has used, which could be legacy Python 2.7 and old libraries. 

For me, one use case is basically when students email me that the HW template code doesn't run for some reason (new programmers are usually tend to forget sending error messages). I then usually ask them to send me the watermark outputs, and often it's because they haven't set their environment correctly and Jupyter Notebook is using a Python 2.7 kernel. 

We may have to drop Python 2.7 support one day, but since Python 2.7 is still the default on many systems, I think it's useful to support it, to see whether someone was running code on Python 2.7 (and then recommend this person to switch to Python >= 3.7). So, I would suggest adding Python 2.7 back to the travis.yml.

Reply:
But someone using Python 2.7 will always be able to get watermark 2.0.2 via pip, so the current functionality will always work, and that will tell you what version of Python, etc they are using.

Reply:
I agree with your points. I was just thinking it might be good to keep 2.7 in the travis CI to see at what point in future 2.7 will not work anymore. I am not suggesting we should actively keep supporting 2.7; it might just be useful to know whether it fails or not.

Reply:
The python_requires setting will stop it installing

Reply:
I've updated this, but Travis hasn't triggered 🤷